### ISSUE TITLE
`DataLossError: corrupted record ..` in `2022-08-03-Creating-TF-Records.ipynb`

### ISSUE THREAD

**Issue by jonathanloganmoran** [jonathanloganmoran/ND0013-Self-Driving-Car-Engineer#1]
### In [`2022-08-03-Creating-TF-Records.ipynb`](https://github.com/jonathanloganmoran/ND0013-Self-Driving-Car-Engineer/blob/293181372e29c744633019a97dcd16e7b9b7ddec/1-Object-Detection-in-Urban-Environments/Exercises/1-3-Creating-TF-Records/2022-08-03-Creating-TF-Records.ipynb):

When I attempt to convert the following `.tfrecord` file into a `TFRecordDataset`:
```
training_segment-1005081002024129653_5313_150_5333_150_with_camera_labels.tfrecord
```
I get the following error message
```
DataLossError: corrupted record at 0 [Op:IteratorGetNext]
```
while iterating over the dataset in a `for` loop and calling on the `open_dataset.Frame()` utility function in the loop body. 

The raw `.tfrecord` file has been provided to me by Udacity.

##### More info:
* The `open_dataset.Frame()` method is a utility function to parse a `.tfrecord` (deserialise its byte string) and return a `Frame` object used by the Waymo Open Dataset.
* The `frames` example file provided by Waymo in their [`waymo-open-dataset/tutorial`](https://github.com/waymo-research/waymo-open-dataset/tree/master/tutorial) repository is able to be converted, parsed and read without errors using the same `for` loop iteration described above.
* The `.tfrecord` file has been downloaded directly from Udacity's servers. I've tried parsing it after conversion from the following formats without success:
    *  the "raw" `.tfrecord` file, uploaded from local machine to Colab instance via GUI, no modifications to `.tfrecord` format before converting to `TFRecordDataset`;
    * the archived tarball file (`data.tgz`), compressed via the MacOS Terminal command `tar -czvf data.tgz data` and uncompressed in Colab with `!tar -xvf data.tgz`.

**Comment by jonathanloganmoran** [comment]
#### **Update 2022-08-08 20:05.15 PST:**
Upon further inspection, it appears that both the `data.tgz` file and its uncompressed source file (the `training_segment-10....tfrecord`) was corrupted prior to uploading to Google Colab. The estimated file size of both the uncompressed record and its compressed tarball file are both showing as ~2 KB, when the true uncompressed file size is ca. 960 MB.

Will try re-downloading the file and repeating the two upload techniques. Thinking of also circumventing the download to local machine step entirely and instead running my `.ipynb` notebook file on the VM that is hosting the source file.

**Comment by jonathanloganmoran** [comment]
**CLOSED 2022-08-08 21:38:30 PST:** Resolved in commit #[9ab7746](https://github.com/jonathanloganmoran/ND0013-Self-Driving-Car-Engineer/commit/9ab774619b268b76d1e3264bb10cc69c3ba7a347).

Current workaround is to download the `.tfrecord` file from GCS bucket directly within Colab using the
```
!gsutil cp gs://{bucket_name}/file_to_download.tfrecord /local_dir/
```
command referenced in the [Snippets: Accesing files](https://colab.research.google.com/notebooks/snippets/accessing_files.ipynb#scrollTo=lPdTf-6O73ll&line=3&uniqifier=1) notebook.


This route requires that the notebook owner has been granted access to the private Waymo Open Dataset GCS bucket and has already authorised their account in Colab by running the following:
```
from google.colab import auth
auth.authenticate_user()
```

Note that only the 'Gsutil URI' via `!gsutil` command and neither the 'Public URL' or 'Authenticated URL' via `!wget` command work in Colab, even when attempting the above authentication.

In the future, it might be wise to avoid moving any large files, e.g., this ~940 MiB `.tfrecord`, between the local machine and Colab instance.
